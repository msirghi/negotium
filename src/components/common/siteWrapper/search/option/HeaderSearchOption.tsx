import { Box } from '@mui/system';
import { Row } from '../../../utilities/row/Row';
import {FC, HTMLAttributes} from 'react';
import { makeStyles } from '@mui/styles';
import colors from '../../../../../common/styles/colors';
import { HeaderSearchOptions } from '../types';

type Props = {
  props: HTMLAttributes<HTMLLIElement>;
  option: HeaderSearchOptions;
};

const useStyles = makeStyles({
  optionTitle: {
    marginLeft: 10,
  },
  optionType: {
    marginLeft: 'auto',
    color: colors.greys['400'],
  },
});

export const HeaderSearchOption: FC<Props> = ({ option, props }) => {
  const classes = useStyles();

  const { title, Icon, id, onClick } = option;
  if (!title) {
    return <div />;
  }
  return (
    // @ts-ignore
    <Box {...props} component={'li'} onClick={onClick} key={id}>
      <Row alignVerticalCenter>
        <Icon color={'primary'} />
        <div className={classes.optionTitle}>{title}</div>
      </Row>
      <div className={classes.optionType}>
        <i>Project</i>
      </div>
    </Box>
  );
};
