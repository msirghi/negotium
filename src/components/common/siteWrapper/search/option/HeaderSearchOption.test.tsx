import CloseIcon from '@mui/icons-material/Close';
import renderer, { act } from 'react-test-renderer';
import { HeaderSearchOption } from './HeaderSearchOption';
import { mount } from 'enzyme';
import { Box } from '@mui/system';

describe('HeaderSearchOption', () => {
  const defaultProps = {
    props: {} as any,
    option: {
      id: '1',
      Icon: CloseIcon,
      title: 'Title',
      onClick: jest.fn(),
    },
  };

  it('should match the snapshot', () => {
    const tree = renderer.create(<HeaderSearchOption {...defaultProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('should return a div if the title is not provided', () => {
    const wrapper = mount(
      <HeaderSearchOption
        {...defaultProps}
        // @ts-ignore
        option={{ ...defaultProps.option, title: undefined }}
      />
    );

    expect(wrapper.find('div')).toHaveLength(1);
  });

  it('should call prop method on box click', () => {
    const wrapper = mount(<HeaderSearchOption {...defaultProps} />);
    const box = wrapper.find(Box);
    act(() => {
      box.simulate('click');
    });
    expect(defaultProps.option.onClick).toBeCalled();
  });
});
