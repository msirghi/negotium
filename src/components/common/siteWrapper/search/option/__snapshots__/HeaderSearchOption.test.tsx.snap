// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSearchOption should match the snapshot 1`] = `
<li
  className="MuiBox-root css-0"
  onClick={[MockFunction]}
>
  <div
    onClick={[Function]}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "width": "initial",
      }
    }
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
      data-testid="CloseIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
    <div
      className="makeStyles-optionTitle-1"
    >
      Title
    </div>
  </div>
  <div
    className="makeStyles-optionType-2"
  >
    <i>
      Project
    </i>
  </div>
</li>
`;
